src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,287 @@
+        logging.info("Mock sending keep active command")
+        # TODO check success
+        #keepActive = await self.default_controller.SendCommand(nodeid=self.dut_node_id, endpoint=kICDBridgedEndpointId, payload=brbinfoCluster.Commands.KeepActive(stayActiveDuration=duration))
+
+    async def _wait_for_active_changed_event(self, timeout) -> int:
+        try:
+            PromisedActiveDuration = self.q.get(block=True, timeout=timeout)
+            logging.info(f"PromisedActiveDuration: {PromisedActiveDuration}")
+            return PromisedActiveDuration
+        except queue.Empty:
+            asserts.assert_true(False, "Timeout on event")
+
+
+
+    @async_test_body
+    async def setup_class(self):
+        super().setup_class()
+        app = os.path.join(pathlib.Path(__file__).resolve().parent, '..','..','out', 'linux-x64-lit-icd', 'lit-icd-app')
+
+        self.kvs = f'kvs_{str(uuid.uuid4())}'
+        self.port = 5543
+        discriminator = random.randint(0, 4095)
+        discriminator = 3850
+        passcode = 20202021
+        app_args = f'--secured-device-port {self.port} --discriminator {discriminator} --passcode {passcode} --KVS {self.kvs} ' + \
+          '--secured-device-port 5545'
+        cmd = f'{app} {app_args}'
+
+        logging.info("Starting ICD Server App")
+        self.app_process = subprocess.Popen(cmd, bufsize=0, shell=True)
+        logging.info("ICD started")
+        time.sleep(3)
+
+        logging.info("Commissioning of ICD to fabric one (TH)")
+        self.icd_nodeid = 1111
+
+        await self.default_controller.CommissionOnNetwork(nodeId=self.icd_nodeid, setupPinCode=passcode, filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=discriminator)
+
+        logging.info("Commissioning of ICD to fabric two (DUT)")
+        params = await self.openCommissioningWindow(dev_ctrl=self.default_controller, node_id=self.icd_nodeid)
+
+        self._ask_for_vendor_commissioniong_ux_operation(params.randomDiscriminator, params.commissioningParameters.setupPinCode)
+
+    def teardown_class(self):
+        logging.warning("Stopping app with SIGTERM")
+        self.app_process.send_signal(signal.SIGTERM.value)
+        test_app_exit_code = self.app_process.wait()
+        os.remove(self.kvs)
+        super().teardown_class()
+
+    #
+    # BRBINFO 4.1 Test Body
+    #
+
+    @async_test_body
+    async def test_TC_BRBINFO_4_1(self):
+
+        icdmCluster = Clusters.Objects.IcdManagement
+        icdmAttributes = icdmCluster.Attributes
+        brbinfoCluster = Clusters.Objects.BridgedDeviceBasicInformation
+        brbinfoAttributes = brbinfoCluster.Attributes
+        basicinfoCluster = Clusters.Objects.BasicInformation
+        basicinfoAttributes = basicinfoCluster.Attributes
+
+        self.endpoint = 0
+
+        # Check commissioned DUT
+        self.step(1)
+        # Confirms commissioning of DUT as it reads basic info from bridge device
```

Comment:
nit, comment say bridged but you are accessing basic info.

I think you want to access the featureMap of the dynamic endpoint representing the ICD device on DUT. If that is the case let me know and I will generate the code I think you will need for that. As we will need that endpoint ahead of time and I think we can add it in the setup section

Reply:
The idea here was simply to access some cluster, like the mandatory basic info just to ensure they are commissioned to the fabric. Changed comments. PTAL